### PR TITLE
Feat(eos_cli_config_gen): add VRRP support for object tracking

### PR DESCRIPTION
**Pull request by ccsnw**
## Change Summary

adding support for object tracking to VRRP config

## Related Issue(s)

Fixes #N/A

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
- Added object tracking support to VRRPs new DM
- Add VRRP Details Documentation
- Remove unnecessary empty space in Documentation
- Fix Vlan110 and Vlan667 typos in Molecule test (ip_address key is missing the underscore)

```
    vrrp_ids:
      - id: < vrid >
        priority_level: < instance_priority >
        advertisement:
          interval: < advertisement_interval>
        preempt:
          enabled: < true | false >
          delay:
            minimum: < integer >
            reload: < integer >
        tracked_object:
          - name: < tracked_object_name_1 >
            decrement: < decrement vrrp priority by 1-254 >
            shutdown: < true | false >
          - name: < tracked_object_name_2 >
            decrement: < decrement vrrp priority by 1-254 >
            shutdown: < true | false >
        ipv4:
          address: < virtual_ip_address >
        ipv6:
          address: < virtual_ip_address >
```

## How to test
added molecule test for vrrp enabled vlan

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by xaviramon** [comment]
Looks good to me, but some points:

1. There is a Data Model change proposed under https://github.com/aristanetworks/ansible-avd/pull/1648. Seems like this one is going to be deprecated on AVD 4.0. I suggest we close this PR and work on the other.
2. In my opinion, we should Document VRRP a little bit better. Now nothing about VRRP is included, at least a priority modification based on object tracking should be added.

**Comment by ccsnw** [comment]
> Looks good to me, but some points:
> 
> 1. There is a Data Model change proposed under [Feat(eos_cli_config_gen): Support multiple VRRP VRID per interface #1648](https://github.com/aristanetworks/ansible-avd/pull/1648). Seems like this one is going to be deprecated on AVD 4.0. I suggest we close this PR and work on the other.
> 2. In my opinion, we should Document VRRP a little bit better. Now nothing about VRRP is included, at least a priority modification based on object tracking should be added.

Hi Xavi,

1) Sounds good - I assume we add the new DM on top of the old one (to keep compatibility)? I can make this available for both models, or just the new one. What do you guys think?

2) Documentation: I can look and expand it. Tackle it in a sep. PR, or include in this?

**Comment by ClausHolbechArista** [comment]
> 1. Sounds good - I assume we add the new DM on top of the old one (to keep compatibility)? I can make this available for both models, or just the new one. What do you guys think?

Just add it to the new model. Let us get the other PR merged in today, so you can rebase and work from that.

> 2. Documentation: I can look and expand it. Tackle it in a sep. PR, or include in this?

Currently we only document the VRRP IPs under each IP interface. You can add a new table to the vlan_interfaces docs with VRRP details. Both the ones you add here, and the existing things. Try to incorperate it into a single table with a row per interface - to avoid it exploding into too long docs.

**Comment by carlbuchmann** [comment]
@ccsnw PR #1648 has been merged with the new vrrp data model.
Please rebase this PR and incorporate your proposed change in the new `vrrp_ids` data model.

**Comment by ccsnw** [comment]
Thanks guys, on it now

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2 [hunk 1]
```diff
@@ -201,6 +201,19 @@ interface {{ vlan_interface }}
 {%             if vrid.ipv6.address is arista.avd.defined %}
    vrrp {{ vrid.id }} ipv6 {{ vrid.ipv6.address }}
 {%             endif %}
+{%             if vrid.tracked_object is arista.avd.defined %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2]: You can delete this line. The next line, you used arista.avd.natural_sort filter, which actually does a defined test. So we can remove this redundant test in this case.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/vlan-interfaces.j2 [hunk 2]
```diff
@@ -71,7 +71,48 @@
...
+{%         endif %}
+{%     endfor %}
+{%     if vlan_interfaces_vrrp_details | length > 0 %}
+
+#### VRRP Details
 
+| Interface | VRRP-ID | Priority | Advertisement Interval | Preempt | Tracked Object Name(s) | Tracked Object Action(s) | IPv4 Virtual IP | IPv6 Virtual IP |
+| --------- | ------- | -------- | ---------------------- | --------| ---------------------- | ------------------------ | --------------- | ----------------|
+{%         for vlan_interface in vlan_interfaces_vrrp_details | arista.avd.natural_sort %}
+{%             for vrid in vlan_interfaces[vlan_interface].vrrp_ids %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/vlan-interfaces.j2]: Need to add a defined test for ```vrid.id```. **for vrid in vlan_interfaces[vlan_interface].vrrp_ids if vrid.id is arista.avd.defined**

Comment (ccsnw) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/vlan-interfaces.j2]: thanks !!

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Thank you for adding documentation for the other VRRP vars.

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**